<a href="https://colab.research.google.com/github/JulianEduardoMendez/programacion1/blob/main/Quiz26Marzo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Julián Eduardo Méndez Poveda // 2260668 // Quiz 26 de Marzo

In [27]:
%%writefile pedirnotas.py
def pedir_nombre():
  return input("Introduce el nombre del estudiante: ")
def pedir_notas():
  n1 = float(input("Nota 1: "))
  n2 = float(input("Nota 2: "))
  n3 = float(input("Nota 3: "))
  n4 = float(input("Nota 4: "))
  return n1, n2, n3, n4

Overwriting pedirnotas.py


In [35]:
%%writefile promedio.py
import statistics
def calcular_promedio(n1, n2, n3, n4):
  return (n1 + n2 + n3 + n4) / 4
def mediana(n1, n2, n3, n4):
  return statistics.median([n1, n2, n3, n4])
def moda(n1, n2, n3, n4):
  return statistics.mode([n1, n2, n3, n4])

Overwriting promedio.py


In [36]:
import pedirnotas
nombre = pedirnotas.pedir_nombre()

Introduce el nombre del estudiante: julian mendez


In [37]:
import importlib
import pedirnotas
importlib.reload(pedirnotas)
importlib.reload(promedio)
notas = pedirnotas.pedir_notas()

Nota 1: 3.0
Nota 2: 3.5
Nota 3: 3.5
Nota 4: 4.0


In [38]:
import promedio
prom = promedio.calcular_promedio(notas[0], notas[1], notas[2], notas[3])
print("Estudiante", nombre)
print("Promedio", prom)
print("mediana", promedio.mediana(notas[0], notas[1], notas[2], notas[3]))
print("moda", promedio.moda(notas[0], notas[1], notas[2], notas[3]))
if prom >= 3:
  print("aprobado")
else:
  print("reprobado")

Estudiante julian mendez
Promedio 3.5
mediana 3.5
moda 3.5
aprobado


El estudiante aprobó la materia, pues su promedio fue mayor a 3 (su promedio fue de 3.5).

La mediana es la nota de la posición de la mitad, pero, en este caso, se promedia las dos notas centrales, y esa será su mediana (3.5).

La moda fue la nota registrada que más se repitió, en este caso es el 3.5.

El desempeño es suficiente pero básico porque, a pesar de que aprobó, lo hizo con un promedio un tanto regular.